streams/src/main/java/org/apache/kafka/streams/state/internals/NamedCache.java

```diff
@@ -182,6 +189,7 @@ synchronized void put(final Bytes key, final LRUCacheEntry value) {
             dirtyKeys.add(key);
         }
         currentSizeBytes += node.size();
+        totalCacheSizeSensor.record(currentSizeBytes);
```

Comment:
Maybe this is a bit paranoid, but I'm a bit worried about a potential performance impact of recording this metric on every cache put/eviction, since `#record` makes a call to the current system time on every invocation (and we've seen issues related to this in the past)

WDYT about splitting out the new metric related changes into a separate PR? I'm fine with still moving forward on that and just watching the benchmarks, I just don't want the poor cache config changes to get reverted again should we happen to find some performance drop. Better to keep things as isolated as possible, even if it's annoying to do/ask for (sorry 😅 )

Reply:
haha i would also agree with you given how my streams PRs have fared. I removed the metrics bit.